### ISSUE TITLE
Setup for the project

### ISSUE THREAD

**Issue by MihaiDavid05** [pidahbus/deep-image-orientation-angle-detection#16]
This is a successful setup for the project. I used:
- Windows 11
- Python 3.9

I first installed Tensorflow following "Step-by-step Instructions" from [this](https://www.tensorflow.org/install/pip). Instead of `pip install "tensorflow<2.11"`, I used `pip install tensorflow==2.8.0`.  (I used the suggested `cudatoolkit=11.2` and `cudnn=8.1.0` )

I then run `pip install numpy==1.23.0` and `pip install protobuf==3.20.*` . I then checked the installation of Tensorflow and it was successful.

I then installed Pytorch using: 
```
pip install torch==1.9.1+cu111 torchvision==0.10.1+cu111 torchaudio==0.9.1 -f https://download.pytorch.org/whl/torch_stable.html   
```

I then installed the rest of the libraries.

My requirements.txt file looks like this:

absl-py==2.1.0
astunparse==1.6.3
cachetools==5.5.0
certifi==2024.8.30
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
filelock==3.15.4
Flask==1.0.2
flatbuffers==24.3.25
fsspec==2024.9.0
gast==0.6.0
google-auth==2.34.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.66.1
h5py==3.11.0
huggingface-hub==0.24.6
idna==3.8
importlib_metadata==8.4.0
itsdangerous==2.2.0
Jinja2==3.1.4
keras==2.8.0
Keras-Preprocessing==1.1.2
libclang==18.1.1
loguru==0.5.3
Markdown==3.7
MarkupSafe==2.1.5
numpy==1.23.0
oauthlib==3.2.2
opencv-python==4.5.5.62
opt-einsum==3.3.0
packaging==24.1
Pillow==9.3.0
protobuf==3.20.3
pyasn1==0.6.0
pyasn1_modules==0.4.0
PyYAML==6.0.2
regex==2024.7.24
requests==2.32.3
requests-oauthlib==2.0.0
rsa==4.9
six==1.16.0
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow==2.8.0
tensorflow-io-gcs-filesystem==0.31.0
termcolor==2.4.0
tf-estimator-nightly==2.8.0.dev2021122109
tokenizers==0.13.3
torch==1.9.1+cu111
torchaudio==0.9.1
torchvision==0.10.1+cu111
tqdm==4.64.0
transformers==4.27.0
typing_extensions==4.12.2
urllib3==2.2.2
Werkzeug==3.0.4
win32-setctime==1.1.0
wrapt==1.16.0
zipp==3.20.1

**Comment by foobarbecue** [comment]
I've been trying various versions of things and getting various errors. Would be really helpful if someone made a dockerfile.

**Comment by Sitronik** [comment]
> I've been trying various versions of things and getting various errors. Would be really helpful if someone made a dockerfile.

Hello everyone, here is the finished result

Dockerfile:
```dockerfile
FROM python:3.9

WORKDIR /app

RUN apt update

RUN apt install ffmpeg libsm6 libxext6 -y

RUN python -m pip install --upgrade pip setuptools wheel

RUN python -m pip install torch==1.8.1 torchvision==0.9.1 torchaudio==0.9.0 -f https://download.pytorch.org/whl/torch_stable.html
RUN python -m pip install torch torchvision torchaudio

COPY . /app

RUN python -m pip install -r /app/requirements.txt

EXPOSE 5050

CMD python /app/run.py
```

requirements.txt:
```
Flask
Pillow == 7.2.0
loguru == 0.5.3
matplotlib
numpy == 1.23.0
pandas
scikit-learn
tensorflow == 2.10.0
tqdm == 4.64.0
transformers == 4.27.0
opencv-python
```
